From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Thu, 19 Mar 2015 16:03:27 +0800
Subject: [PATCH] {BUG} drivers/net/ethernet/qlogic/qla3xxx: fix constant
 overflow

ISP_NVRAM_MASK = (0x000F << 16),
ISP_NVRAM_MASK << 16 overflows 32 bits...

Fixes 5a4faa873782d748960b02fdec95e3d4d2e3ae38

Reported: https://lkml.org/lkml/2017/1/15/54
---
 drivers/net/ethernet/qlogic/qla3xxx.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/qlogic/qla3xxx.c b/drivers/net/ethernet/qlogic/qla3xxx.c
index 9e5264d8773b..cf1299fa6122 100644
--- a/drivers/net/ethernet/qlogic/qla3xxx.c
+++ b/drivers/net/ethernet/qlogic/qla3xxx.c
@@ -380,8 +380,7 @@ static void fm93c56a_select(struct ql3_adapter *qdev)
 
 	qdev->eeprom_cmd_data = AUBURN_EEPROM_CS_1;
 	ql_write_nvram_reg(qdev, spir, ISP_NVRAM_MASK | qdev->eeprom_cmd_data);
-	ql_write_nvram_reg(qdev, spir,
-			   ((ISP_NVRAM_MASK << 16) | qdev->eeprom_cmd_data));
+	ql_write_nvram_reg(qdev, spir, qdev->eeprom_cmd_data);
 }
 
 /*
-- 
